common/src/test/scala/explore/utils/StreamSpec.scala

```diff
@@ -58,3 +58,23 @@ class StreamSpec extends munit.CatsEffectSuite:
 
     TestControl.executeEmbed(program).map((a, b) => assertNotEquals(a, b))
   }
+
+  test("keyedSwitchEvalMap cancels selectively") {
+    val stream                 = Stream(1, 2, 3).evalMap(IO.sleep(0.25.seconds).as(_))
```

Comment:
```suggestion
    val stream                 = Stream(1, 2, 3).meteredStartImmediately(0.25.seconds)
```
I think this does the same